### PR TITLE
Fix ARN reference contributor returning duplicate entries

### PR DESCRIPTION
**Pull request by rli**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt [hunk 1]
```diff
@@ -17,9 +18,15 @@ class ArnPsiReferenceContributor : PsiReferenceContributor() {
             ) {
                 override fun accepts(o: Any?, context: ProcessingContext): Boolean {
                     if (o == null || o !is PsiElement) return false
+                    val manipulator = ElementManipulators.getManipulator(o)
+                    if (manipulator == null) return false
 
-                    if (o.text.contains("arn:")) {
-                        return true
+                    try {
+                        if (manipulator.getRangeInElement(o).substring(o.text).contains("arn:")) {
```
Comment (breedloj) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt:25]: Can we not just retrieve the `TextRange` directly off the `PsiElement`?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt [hunk 2]
```diff
@@ -17,9 +18,15 @@ class ArnPsiReferenceContributor : PsiReferenceContributor() {
...
+                            return true
+                        }
+                    } catch (e: Exception) {
+                        return false
                     }
```
Comment (breedloj) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt:30]: Avoidable by checking the element's text length up front?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt (hunk 1 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt:25]: the manipulators are language-aware and will strip out text that is not relevant to us (for example, stripping quotes in literal declarations, or only no-oping if we're not looking at a leaf element in the YAML case)

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt (hunk 2 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt:30]: will keep the try/catch, but add explicit element length check

**Review by breedloj — APPROVED** [review]

**Pull request merged**